src/Compilers/Core/Portable/Diagnostic/DiagnosticFormatter.cs

```diff
@@ -101,6 +109,34 @@ internal string GetMessagePrefix(Diagnostic diagnostic, CultureInfo culture)
                 diagnostic.Id);
         }
 
-        internal static readonly DiagnosticFormatter Instance = new DiagnosticFormatter();
+        internal string Format(Diagnostic diagnostic, IFormatProvider formatter, AnalyzerManager analyzerManager)
+        {
+#pragma warning disable RS0013 // we do need descriptor instance here
+            if (!_displayAnalyzer || analyzerManager == null || diagnostic.Descriptor == null)
+            {
+                return Format(diagnostic, formatter);
+            }
+
+            var analyzerIdentity = GetAnalyzerIdentity(analyzerManager.GetDiagnosticAnalyzer(diagnostic.Descriptor));
+            if (analyzerIdentity == null)
+            {
+                return Format(diagnostic, formatter);
+            }
+
+            return Format(diagnostic, formatter) + ": " + analyzerIdentity;
```

Comment:
not sure whether this is right format. using other delimiter that makes parsing easier might be better.